src/lib/spinel/radio_spinel.hpp

```diff
@@ -1216,6 +1241,41 @@ class RadioSpinel : private Logger
 
     void HandleCompatibilityError(void);
 
+    class Metrics : public otRadioSpinelMetrics
+    {
+    public:
+        Metrics(RadioSpinelCallbacks &aCallbacks)
+            : mCallbacks(aCallbacks)
+        {
+            memset(reinterpret_cast<void *>(this), 0, sizeof(otRadioSpinelMetrics));
```

Comment:
Is reinterpret_cast required? I see some memset with a pointer type not being `void *` under src/lib.

Reply:
Yes, it is required. Otherwise, it reports the following error:

```
error: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘class ot::Spinel::RadioSpinel::Metrics’ with no trivial copy-assignment [-Werror=class-memaccess]
```